src/quaternion_maps.jl

```diff
@@ -0,0 +1,256 @@
+
+abstract type VectorPart <: QuatMap end
+abstract type ExponentialMap <: QuatMap end
+abstract type MRPMap <: QuatMap end
+abstract type CayleyMap <: QuatMap end
+abstract type IdentityMap <: QuatMap end
```

Comment:
How about making these concrete singleton instances rather than types?

That will avoid a lot of `::Type` everywhere, and also fit the expectation that calling a type will actually construct something of that type!

Another idea would be to do away with `QuatMap` entirely (unless you need it for dispatch), and to do something like

```julia
function mrp_qmap(v)
    ...
    UnitQuaternion{MRPMap}(...)
end
const MRPMap = typeof(mrp_qmap)
```

I'm not quite sure it makes sense, just throwing it out there. Currently I feel a little confusion with the way this code is arranged.

Reply:
I went with your suggestion of making them concrete singletons. I'm not sure I like the syntax `ExponentialMap()(q)` but in practice I don't think this syntax will be exposed to the user, since it's tucked away inside of the new `RotationError` I've defined.

Reply:
The new interface eliminates using types as functions and uses `forward_map(emap, v)` and `inverse_map(emap, q)` instead.

Reply:
> I'm not sure I like the syntax `ExponentialMap()(q)`

Agreed!  However if we did need to expose this to the user there's a very simple solution, which again is taking the idea that these are "just functions" to the logical endpoint: just define

```
const expmap = ExponentialMap()

export expmap
```

Then the user does `expmap(q)` and `inv(expmap)(q)`. This is very similar to how functions work internally in the Julia compiler. (Actually it's basically the same, with each function getting a unique type which is a subtype of `Function`.)

For now though, I like that you've kept this internal to the package.